components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/APIDefinition.java

```diff
@@ -133,7 +133,7 @@
      * @param returnJsonContent whether to return the converted json form of the OpenAPI definition
      * @return APIDefinitionValidationResponse object with validation information
      */
-    public abstract APIDefinitionValidationResponse validateAPIDefinition(String apiDefinition,
+    public abstract APIDefinitionValidationResponse validateAPIDefinition(String apiDefinition, String url,
```

Comment:
Shall we have two overloaded method for this? - original one and the one added with url.
From the implementation level, we can only implement the method with the URL (as you have done already).
From the other method (without URL) -> call the one with URL with passing "" as URL.

In all the instances that you have used `validateAPIDefinition(schemaToBeValidated, "", returnJSONContent);` use the method without URL instead.

Reply:
Fixed with 390d69634ebee69b36035e9a47e7b6ac570f0df3